src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -724,6 +807,279 @@ public final void markDirty()
     }
 
     /**
+     * Add a new container to the building.
+     * @param pos position to add.
+     */
+    public void addContainerPosition(BlockPos pos)
+    {
+        containerList.add(pos);
+    }
+
+    /**
+     * Remove a container from the building.
+     * @param pos position to remove.
+     */
+    public void removeContainerPosition(BlockPos pos)
+    {
+        containerList.remove(pos);
+    }
+
+    /**
+     * Get all additional containers which belong to the building.
+     * @return a unmodifiable list of the container positions.
+     */
+    public List<BlockPos> getAdditionalCountainers()
+    {
+        return Collections.unmodifiableList(containerList);
+    }
+
+    //------------------------- Starting Required Tools/Item handling -------------------------//
+
+    /**
+     * Override this method if you want to keep an amount of items in inventory.
+     * When the inventory is full, everything get's dumped into the building chest.
+     * But you can use this method to hold some stacks back.
+     *
+     * @return a list of objects which should be kept.
+     */
+    public Map<ItemStorage, Integer> needXForWorker()
+    {
+        return new HashMap<>();
+    }
+
+    /**
+     * Check if the building is receiving the required items.
+     * @return true if so.
+     */
+    public boolean hasOnGoingDelivery()
+    {
+        return onGoingDelivery;
+    }
+
+    /**
+     * Check if the building is receiving the required items.
+     * @param valueToSet true or false
+     */
+    public void setOnGoingDelivery(boolean valueToSet)
+    {
+        this.onGoingDelivery = valueToSet;
+    }
+
+    /**
+     * Check if the worker needs anything. Tool or item.
+     * @return true if so.
+     */
+    public boolean needsAnything()
+    {
+        return !itemsCurrentlyNeeded.isEmpty() || needsShovel || needsAxe || needsHoe || needsWeapon || needsPickaxe;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (5) used in the expression (maximum allowed 4). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)